### DISCUSSION TITLE
ROS2 Humble FastRTPS communication between Docker Ubuntu container in Windows host and a native Ubuntu Machine.

### DISCUSSION THREAD

**Discussion by applebro007** [eProsima/Fast-DDS#3637] [Q&A]
I have the following setup:
- Linux Docker running with ROS2 humble installed. This docker runs on Windows 11 host (Home Edition).
- This Windows 11 host also has ROS2 humble natively installed.
- Linux Machine with ROS2 humble installed natively.

What I am trying to achieve (by any hook or crook): 
- Multicast communication between ROS2 in the docker and ROS2 in the Linux machine.
- Multicast communication between ROS2 in the docker and ROS2 in Windows 11 host machine.
- Unicast (Publisher/Subscriber) communication between ROS2 in the docker and ROS2 in the Linux machine.

I have used the default settings for ROS2 - FastDDS itself. As a note, I have tried this same docker in a Linux machine (as host) and it works simply with --net=host in docker run command. However, from the Docker community I confirmed that --net=host does not work in Windows due to the WSL2 virtualization that NATs the ip. 

Is there a way to use UDP or TCP protocol to establish communication through FastDDS ? If not, has anyone found a workaround for this by any other means? I'll be happy to buy you a coffee :)

**Comment by applebro007** [comment]
Hi,
If you require any additional information on my setup and my requirement, please do not hesitate to ask and I'll be happy to provide you.
I have attempted the above with several methods, but perhaps due to my lack of experience with DDS, I am unable to reach a successful solution. If you could guide me on what I should try out (even if not a certain solution), please feel free.
Thanks

**Comment by JLBuenoLopez** [comment]
Hi @Paulvp,

I have also read your question in [ROS answers](https://answers.ros.org/question/416676/ros2-multicast-in-docker-running-on-windows/) which have enlightened me a bit. It seems that your issue is that you are not receiving the `Hello World` message when using the `ros2multicast` tool. This is a debugging tool which does not send using the DDS middleware. It is only checking that multicast traffic is coming through. You might find interesting this [other question](https://superuser.com/questions/1762729/multicast-from-wsl2-to-win-does-not-work) I have found in the internet and that it seems to be related with your issue. Therefore, it seems that this is unrelated to Fast DDS library and more related to the network infrastructure not supporting multicast communication.

A possible workaround you might consider is using [initial peers](https://fast-dds.docs.eprosima.com/en/latest/fastdds/discovery/simple.html#simple-initial-peers) so you can [disable the multicast traffic](https://fast-dds.docs.eprosima.com/en/latest/fastdds/transport/disabling_multicast.html#disabling-all-multicast-traffic).
